common/eth2/src/types.rs

```diff
@@ -189,6 +189,14 @@ impl fmt::Display for StateId {
 #[serde(bound = "T: Serialize + serde::de::DeserializeOwned")]
 pub struct DutiesResponse<T: Serialize + serde::de::DeserializeOwned> {
     pub dependent_root: Hash256,
+    pub execution_optimistic: bool,
```

Comment:
Good point.
It does not work in this case, the VC fails with:
```
RequestFailed(Reqwest(reqwest::Error { kind: Decode, source: Error("missing field `execution_optimistic`", line: 1, column: 482) }))
```

Reply:
Breaking VC changes are usually pretty annoying to deal with, we could add a field default for the case where it's missing?

I'm tempted to say we default it to `true`, so that an updated VC on a merge network stalls against an old BN and forces the updated. On mainnet hopefully we don't check `execution_optimistic`?

Reply:
Found it, it's in the `beacon_chain` crate when fetching attestations: https://github.com/sigp/lighthouse/pull/3140

Reply:
Attestation filtering was done here: https://github.com/sigp/lighthouse/pull/3140
Sync committee filtering is a WIP: https://github.com/sigp/lighthouse/pull/3191

Seems like it is enforced on the BN side?

Reply:
A default seems pretty safe then, or maybe an `Option<bool>` is better. The VC should be able to deserialize an `Option<bool>` to `None` in the case where it's missing

Reply:
The VC also requests Pubkey -> Val index during boot:
https://github.com/macladson/lighthouse/blob/ex-opt-api/validator_client/src/duties_service.rs#L390-L393

That endpoint gets deserialized as `ExecutionOptimisticResponse`. I'm not sure but setting `#[serde(default)]`  on that field may have broader implications than just adding it to `DutiesResponse` would.

Reply:
I decided to go with an `Option<bool>` for all of the `ExecutionOptimistic` types. Based on my testing this supports backwards compatibility with older BNs.